frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/useInitialAutoLayout.ts

```diff
@@ -26,6 +31,14 @@ const getHiddenNodeIdsFromUrl = async (): Promise<string[]> => {
   return hiddenNodeIds ? hiddenNodeIds.split(',') : []
 }
 
+const getShowModeFromUrl = (): string | undefined => {
```

Comment:
If it is done this way, the cast by as in line 84 should be eliminated:

```suggestion
const getShowModeFromUrl = (): ShowMode | undefined => {
```

Reply:
Thank you for your feedback. You're absolutely right.
I've made the following adjustments, but I've also added some code to validate the received showMode. Do you think it's a bit redundant?

https://github.com/liam-hq/liam/pull/452/commits/0890fafc470fb8f7501fdb7e3a14d838592fc8b5


```ts
const getShowModeFromUrl = (): ShowMode => {
  const urlParams = new URLSearchParams(window.location.search)
  const showModeQueryParam: QueryParam = 'showMode'
  const showMode = urlParams.get(showModeQueryParam)

  if (!showMode) {
    return 'TABLE_NAME'
  }

  const result = v.safeParse(showModeSchema, showMode)
  if (result.success && result.output) {
    return result.output
  }

  return 'TABLE_NAME'
}

```